src/python/CRABClient/Commands/setfilestatus.py

```diff
@@ -0,0 +1,156 @@
+
+        # intitalize, and validate args
+        instance = self.options.instance
+        dataset = self.options.dataset
+        files = self.options.files
+        status = self.options.status
+        self.logger.debug('instance     = %s' % instance)
+        self.logger.debug('dataset      = %s' % dataset)
+        self.logger.debug('files        = %s' % files)
+        self.logger.debug('status       = %s' % status)
+
+        statusToSet = 1 if status == 'VALID' else 0
+
+        filesToChange = None
+        if files:
+            # did the user specify the name of a file containing a list of LFN's ?
+            try:
+                with open(files, 'r') as f:
+                    flist = [lfn.strip() for lfn in f]
+                    filesToChange = ','.join(flist)
+            except IOError:
+                # no. Assume we have a comma separated list of LFN's (a single LFN is also OK)
+                filesToChange = files.strip(",").strip()
+            finally:
+                # files and dataset options are mutually exclusive
+                dataset = None
+            if ',' in filesToChange:
+                raise NotImplementedError('list of LFNs is not supported yet')
+
+        # from DBS instance, to DBS REST services
+        dbsReader, dbsWriter = getDbsREST(instance=instance, logger=self.logger,
+                                          cert=self.proxyfilename, key=self.proxyfilename,
+                                          version=__version__)
+        # we will need the dataset name
+        if dataset:
+            datasetName = dataset
+        else:
+            # get it from DBS
+            lfn = filesToChange.split(',')[0]
+            query = {'logical_file_name': lfn}
+            out, rc, msg = dbsReader.get(uri='datasets', data=query)
+            if not out:
+                self.logger.error("ERROR: file %s not found in DBS" % lfn)
+                raise ConfigurationException
+            datasetName = out[0]['dataset']
+            self.logger.info('LFN to be changed belongs to dataset %s' % datasetName)
+
+        # when acting on a list of LFN's, can't print status of all files before/after
+        # best we can do is to print the number of valid/invalid file in the dataset
+        # before/after.
+
+        self.logFilesTally(dataset=datasetName, dbs=dbsReader)
+
+        if filesToChange:
+            data = {'logical_file_name': filesToChange, 'is_file_valid': statusToSet}
+        if dataset:
+            data = {'dataset': dataset, 'is_file_valid': statusToSet}
+        jdata = json.dumps(data)  # PUT requires data in JSON format
+        out, rc, msg = dbsWriter.put(uri='files', data=jdata)
+        if rc == 200 and msg == 'OK':
+            self.logger.info("File(s) status changed successfully")
+            result = 'SUCCESS'
+        else:
+            msg = "File(s) status change failed: %s" % out
+            raise CommandFailedException(msg)
```

Comment:
This would have better fit #5225 !
The only reasons I had to use both is that e.g. users of `status` API are used to always get a dictionary back, so I do not want to suddenly raise exceptions. In other cases, esp. where commands was already capable of rising exceptions, I thought that raising was making for cleaner code. Fewer "if".

Reply:
I do not have any suggestion for this. I just want to know and never notice when I reviewed https://github.com/dmwm/CRABClient/pull/5225.
Thanks for explanation.